### PR TITLE
feat(unlock-js, locksmith): updated UnlockJS to support transaction options

### PR DESCRIPTION
**Pull request by julien51**
# Description

I am adding a new critical feature to the unlock-js library: the ability to pass transaction options.

This will be used to run "simulations" that can then help us do 2 things:
* account for real gas costs on renewals or credit card purchases (rather than hardcoded value)
* not showing the "claim" button when the key could not be airdropped because the lock uses a hook

# Checklist:

- [X] 1 PR, 1 purpose: my Pull Request applies to a single purpose
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have updated the docs to reflect my changes if applicable
- [ ] I have added tests (and stories for frontend components) that prove my fix is effective or that my feature works
- [ ] I have performed a self-review of my own code
- [ ] If my code involves visual changes, I am adding applicable screenshots to this thread

### CODE REVIEW

File: packages/unlock-js/package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "@unlock-protocol/unlock-js",
-  "version": "0.26.8",
+  "version": "0.27.0",
```
Comment (julien51) [inline packages/unlock-js/package.json:3]: Since it changes the API, I am bumping minor

**Comment by julien51** [comment]
@clemsos what's the way to run all tests again?

**Comment by clemsos** [comment]
> /run-all-tests

so this works but the results aren't shown here :/ 

you have to dig into the "actions" tab to find it > https://github.com/unlock-protocol/unlock/actions/runs/3086054579

there is an open issue for this already but no fix in mind so far #9306
should be a way with existing gh actions

**Comment by julien51** [comment]
HA! Thank you for this. I am _very_ surprised that tests are passing here because I changed the API for `walletService` on many calls and I would expect them to fail as a consequence :(

**Comment by searchableguy** [comment]
The code itself looks good (It does change the grantKeyExtension and grantKeys in dispatcher we depend on). I tested bunch of functions on the dashboard too.

**Review by searchableguy — APPROVED** [review]
I'm guessing the transaction object isn't used intentionally in most cases in this PR.

**Pull request merged**